mulog-kinesis/src/com/brunobonacci/mulog/publishers/aws_utils.clj

```diff
@@ -0,0 +1,22 @@
+(ns com.brunobonacci.mulog.publishers.aws-utils
+  (:require
+    [cognitect.aws.client.api :as aws]))
+
+(defn create-kinesis-client
+  [params]
+  (aws/client params))
+
+(defn publish!
+  [kinesis-client stream-name records]
+  (let [rsp (aws/invoke kinesis-client {:op      :PutRecords
+                                        :request {:StreamName stream-name
+                                                  :Records    records}})
+        failed-record-count (:FailedRecordCount rsp)]
+    (if (or (nil? failed-record-count)
+          (not (zero? failed-record-count)))
+      (println
+        (format "mu/log kinesis stream '%s' publisher failure; Reason: %s" stream-name rsp)))))
```

Comment:
In this case it is better to throw an exception so that the request will be retried in a later time.
As it is, if the request fails, the events are lost

Reply:
Agree, my initial implementation used that approach and for the responses such as 

```
 {:FailedRecordCount 2,
  ; :Records [{:SequenceNumber "49606637818941416612640445792517439640700155362084388866", :ShardId "shardId-000000000000"}]}
```
it absolutely makes sense.
However, if the response is 
```
  {:__type "ResourceNotFoundException",
   :message "Stream KINESIS-LOCAL-STREAM-NAME under account 000000000000 not found.",
   :cognitect.anomalies/category :cognitect.anomalies/incorrect}
```
should I still throw ex. or logging would be enough?

Reply:
so `:cognitect.anomalies/category` is a better way to determine if there was a failure of not...
but the point is what will you do in case of failure?
there is nothing you can do.
So the only case i would check is for a successful sent in which case you can discard the elements from the buffer.